src/utils/floating-ui.ts

```diff
@@ -399,12 +440,22 @@ export async function positionFloatingUI({
     })
   });
 
-  if (middlewareData?.arrow) {
-    const { x: arrowX, y: arrowY } = middlewareData.arrow;
+  if (arrowEl && middlewareData?.arrow) {
```

Comment:
No need for optional chaining on `middlewareData` as [it is guaranteed](https://github.com/floating-ui/floating-ui/blob/master/packages/core/src/types.ts#L87).